addons/pos_stripe/models/pos_payment_method.py

```diff
@@ -96,11 +96,13 @@ def stripe_payment_intent(self, amount):
         try:
             data = werkzeug.urls.url_encode(params)
             resp = requests.post(endpoint, data=data, auth=(self.sudo()._get_stripe_secret_key(), ''), timeout=TIMEOUT)
+            resp = resp.json()
+            _logger.info("Stripe payment intent response: %s", resp)
```

Comment:
Be careful not to display private information in the server logs.
name or credit card number

Reply:
I doubt Stripe will provide such details into an API calls this request also happen when sending the request to the terminal (from what I tested), at this point it does not know anything on the customer or card

Reply:
and about `clientSecret` which will no longer be so secret

Reply:
It will be very strange to me to see that stripe return this kind of information in a response, but if yes, good catch!

Reply:
```
client_secret
nullable string
retrievable with publishable key
The client secret of this PaymentIntent. Used for client-side retrieval using a publishable key.

The client secret can be used to complete a payment from your frontend. It should not be stored, logged, or exposed to anyone other than the customer. Make sure that you have TLS enabled on any page that includes the client secret.
```

Reply:
t should not be stored, logged, or exposed to anyone

Reply:
https://docs.stripe.com/api/payment_intents/object#payment_intent_object-client_secret

Reply:
@qle-odoo good question! 

I noticed that it changed each time so I don't know if it can really be use for bad intentions.
Also, the people able to read the logs are the one that are already granted admins access to the database, so they can simply steal Stripe credentials if they have malicious ideas

Here are example of request response as example
```js
{'id': 'pi_3P4K0SFYmFWud15u1NY85lIA', 'object': 'payment_intent', 'amount': 750, 'amount_capturable': 0, 'amount_details': {'tip': {}}, 'amount_received': 0, 'application': None, 'application_fee_amount': None, 'automatic_payment_methods': None, 'canceled_at': None, 'cancellation_reason': None, 'capture_method': 'manual', 'client_secret': 'pi_3P4K0SFYmFWud15u1NY85lIA_secret_SesHMOTBaIQiDg0AjJKxIwH1J', 'confirmation_method': 'automatic', 'created': 1712828000, 'currency': 'eur', 'customer': None, 'description': None, 'invoice': None, 'last_payment_error': None, 'latest_charge': None, 'livemode': False, 'metadata': {}, 'next_action': None, 'on_behalf_of': None, 'payment_method': None, 'payment_method_configuration_details': None, 'payment_method_options': {'card_present': {'request_extended_authorization': False, 'request_incremental_authorization_support': False}}, 'payment_method_types': ['card_present'], 'processing': None, 'receipt_email': None, 'review': None, 'setup_future_usage': None, 'shipping': None, 'source': None, 'statement_descriptor': None, 'statement_descriptor_suffix': None, 'status': 'requires_payment_method', 'transfer_data': None, 'transfer_group': None} 
```

```js
{'id': 'pi_3P4K4eFYmFWud15u03vAnbSy', 'object': 'payment_intent', 'amount': 750, 'amount_capturable': 0, 'amount_details': {'tip': {}}, 'amount_received': 0, 'application': None, 'application_fee_amount': None, 'automatic_payment_methods': None, 'canceled_at': None, 'cancellation_reason': None, 'capture_method': 'manual', 'client_secret': 'pi_3P4K4eFYmFWud15u03vAnbSy_secret_XoZCOdUbswa6z3e0FzeHsN0g8', 'confirmation_method': 'automatic', 'created': 1712828260, 'currency': 'eur', 'customer': None, 'description': None, 'invoice': None, 'last_payment_error': None, 'latest_charge': None, 'livemode': False, 'metadata': {}, 'next_action': None, 'on_behalf_of': None, 'payment_method': None, 'payment_method_configuration_details': None, 'payment_method_options': {'card_present': {'request_extended_authorization': False, 'request_incremental_authorization_support': False}}, 'payment_method_types': ['card_present'], 'processing': None, 'receipt_email': None, 'review': None, 'setup_future_usage': None, 'shipping': None, 'source': None, 'statement_descriptor': None, 'statement_descriptor_suffix': None, 'status': 'requires_payment_method', 'transfer_data': None, 'transfer_group': None} 
```

Reply:
> it should not be stored, logged, or exposed to anyone

Then it's already "exposed" as we can read it from the network Preview tab on the PoS :thinking: 
![image](https://github.com/odoo/odoo/assets/60775325/451e2dc9-a03d-4d89-986b-85a40622af9f)

Reply:
> it should not be stored, logged, or exposed to anyone

> It should not be stored, logged, or exposed to anyone **other than the customer.**

Reply:
Yes if you are onsite.
With this secret you can refund the linked transaction.
And therefore those who see the logs would have the means to make a refund.
Without necessarily being an admin...
the question is:
What is the desired objective of logging this specific information?
Is the risk necessary?

Reply:
@qle-odoo ok I do agree on this last point :) I'll remove this part of the PR

Reply:
Lol maybe that's why you didn't want security to see your PR? :)

Reply:
> @qle-odoo ok I do agree on this last point :) I'll remove this part of the PR

You can keep the logger but without critical infos

Reply:
they are likely redundant on what the customer have on their Stripe portal so not worth the risk I would say

Reply:
@qle-odoo I edited the code to redacted the secret value, is there any other which looks sensible to you ?